egs/babel_multilang/s5/local/nnet3/run_tdnn_multilingual.sh

```diff
@@ -0,0 +1,296 @@
+. local.conf || exit 1;
+
+
+num_langs=${#lang_list[@]}
+
+echo "$0 $@"  # Print the command line for logging
+if ! cuda-compiled; then
+  cat <<EOF && exit 1
+This script is intended to be used with GPUs but you have not compiled Kaldi with CUDA
+If you want to use GPUs (and have them), go to src/, and configure and make on a machine
+where "nvcc" is installed.
+EOF
+fi
+
+for lang_index in `seq 0 $[$num_langs-1]`; do
+  for f in data/${lang_list[$lang_index]}/train/{feats.scp,text} exp/${lang_list[$lang_index]}/$alidir/ali.1.gz exp/${lang_list[$lang_index]}/$alidir/tree; do
+    [ ! -f $f ] && echo "$0: no such file $f" && exit 1;
+  done
+done
+
+if [ "$speed_perturb" == "true" ]; then suffix=_sp; fi
+dir=${dir}${suffix}
+
+if $use_pitch; then feat_suffix=${feat_suffix}_pitch ; fi
+
+for lang_index in `seq 0 $[$num_langs-1]`; do
+  echo "$0: extract high resolution 40dim MFCC + pitch for speed-perturbed data "
+  echo "and extract alignment."
+  local/nnet3/run_common_langs.sh --stage $stage \
+    --use-pitch $use_pitch \
+    --speed-perturb $speed_perturb ${lang_list[$lang_index]} || exit 1;
+done
+
+if $use_ivector; then
+  ivector_suffix=
+  if [ -z "$ivector_extractor" ]; then
+    mkdir -p data/multi
+    mkdir -p exp/multi/nnet3
+    global_extractor=exp/multi/nnet3
+    ivector_extractor=$global_extractor/extractor
+    multi_data_dir=data/multi/train${suffix}_hires
+    ivector_suffix=_gb
+    echo "$0: combine training data using all langs for training global i-vector extractor."
+    if [ ! -f $multi_data_dir/.done ]; then
+      echo ---------------------------------------------------------------------
+      echo "Pooling training data in $multi_data_dir on" `date`
+      echo ---------------------------------------------------------------------
+      mkdir -p $multi_data_dir
+      combine_lang_list=""
+      for lang_index in `seq 0 $[$num_langs-1]`;do
+        combine_lang_list="$combine_lang_list data/${lang_list[$lang_index]}/train${suffix}_hires"
+      done
+      utils/combine_data.sh data/multi/train${suffix}_hires $combine_lang_list
+      utils/validate_data_dir.sh --no-feats data/multi/train${suffix}_hires
+      touch data/multi/train${suffix}_hires/.done
+    fi
+  fi
+  if [ ! -f $global_extractor/extractor/.done ]; then
+    if [ -z $lda_mllt_lang ]; then lda_mllt_lang=${lang_list[0]}; fi
+    echo "$0: Generate global i-vector extractor on pooled data from all "
+    echo "languages in $multi_data_dir, using an LDA+MLLT transform trained "
+    echo "on ${lda_mllt_lang}."
+    local/nnet3/run_shared_ivector_extractor.sh  \
+      --suffix $suffix \
+      --stage $stage $lda_mllt_lang \
+      $multi_data_dir $global_extractor || exit 1;
+    touch $global_extractor/extractor/.done
+  fi
+  echo "$0: Extracts ivector for all languages using $global_extractor/extractor."
+  for lang_index in `seq 0 $[$num_langs-1]`; do
+    local/nnet3/extract_ivector_lang.sh --stage $stage \
+      --train-set train$suffix \
```

Comment:
At this time, I think I've diverged too much from the changes on this PR so I think there is no point for me to continue running the scripts. Please try to address the rest of the comments ASAP so that I can start testing again. Also, try to prioritize this above your other work, as it would be good if we finalize this (and we are planning to use it in some other projects + I'm getting asked constantly if it's ready yet). Thanks